ml-agents/mlagents/trainers/agent_processor.py

```diff
@@ -258,3 +258,8 @@ def __init__(
             self.behavior_id
         )
         self.publish_trajectory_queue(self.trajectory_queue)
+
+    def set_environment_stats(self, env_stats: Dict[str, float]) -> None:
```

Comment:
docstring on this method (even though it is not public)
Is this method only ever going to be used with TensorBoardStats ? If it is the case, the name should reflect that. If not, add a comment explaining what should call this method.

Reply:
Can we call this method `record_environment_stats`? Since under the hood we might be doing both `set` and `add` for different stats

Reply:
> Is this method only ever going to be used with TensorBoardStats
I calls the StatsReporter, which writes to multiple locations, but I think TensorBoard is the main usecase.

Renamed to record_environment_stats() and added docstring